src/main/java/org/jfrog/hudson/pipeline/common/Utils.java

```diff
@@ -270,17 +270,18 @@ public String call() throws IOException {
      * @param ws       - The workspace
      */
     public static void launch(String taskName, Launcher launcher, ArgumentListBuilder args, EnvVars env, TaskListener listener, FilePath ws) {
-        boolean failed;
+
         try {
             int exitValue = launcher.launch().cmds(args).envs(env).stdout(listener).stderr(listener.getLogger()).pwd(ws).join();
-            failed = (exitValue != 0);
+            if (exitValue != 0) {
+                throw new RuntimeException(taskName + " build failed with exit code " + exitValue);
+            }
         } catch (Exception e) {
-            listener.error("Couldn't execute " + taskName + " task. " + ExceptionUtils.getMessage(e));
-            failed = true;
-        }
-        if (failed) {
-            throw new RuntimeException(taskName + " build failed");
+            String errorMessage = "Couldn't execute " + taskName + " task. " + ExceptionUtils.getMessage(e);
```

Comment:
I think getRootCauseMessage is better since it prints the first message in the stack trace (the actual reason). See explanation here: http://learningviacode.blogspot.com/2013/06/exceptionutils.html
```suggestion
            String errorMessage = "Couldn't execute " + taskName + " task. " + ExceptionUtils.getRootCauseMessage(e);
```